spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkWriteConf.java

```diff
@@ -164,6 +182,245 @@ public void testSparkWriteConfDistributionModeWithEverything() {
         });
   }
 
+  @Test
+  public void testSparkWriteConfWriteProperties() {
+    Object[][] propertiesSuites =
+        new Object[][] {
+          {
+            ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_LEVEL, "3"),
+            ImmutableMap.of(
+                DEFAULT_FILE_FORMAT,
+                "parquet",
+                DELETE_DEFAULT_FILE_FORMAT,
+                "parquet",
+                TableProperties.PARQUET_COMPRESSION,
+                "gzip",
+                TableProperties.DELETE_PARQUET_COMPRESSION,
+                "snappy"),
+            ImmutableMap.of(
+                DELETE_PARQUET_COMPRESSION,
+                "zstd",
+                PARQUET_COMPRESSION,
+                "zstd",
+                PARQUET_COMPRESSION_LEVEL,
+                "3",
+                DELETE_PARQUET_COMPRESSION_LEVEL,
+                "3")
+          },
+          {
+            ImmutableMap.of(
+                COMPRESSION_CODEC, "zstd", SparkSQLProperties.COMPRESSION_STRATEGY, "compression"),
+            ImmutableMap.of(
+                DEFAULT_FILE_FORMAT,
+                "orc",
+                DELETE_DEFAULT_FILE_FORMAT,
+                "orc",
+                ORC_COMPRESSION,
+                "zlib",
+                DELETE_ORC_COMPRESSION,
+                "snappy"),
+            ImmutableMap.of(
+                DELETE_ORC_COMPRESSION,
+                "zstd",
+                ORC_COMPRESSION,
+                "zstd",
+                DELETE_ORC_COMPRESSION_STRATEGY,
+                "compression",
+                ORC_COMPRESSION_STRATEGY,
+                "compression")
+          },
+          {
+            ImmutableMap.of(COMPRESSION_CODEC, "zstd", COMPRESSION_LEVEL, "9"),
+            ImmutableMap.of(
+                DEFAULT_FILE_FORMAT,
+                "avro",
+                DELETE_DEFAULT_FILE_FORMAT,
+                "avro",
+                AVRO_COMPRESSION,
+                "gzip",
+                DELETE_AVRO_COMPRESSION,
+                "snappy"),
+            ImmutableMap.of(
+                DELETE_AVRO_COMPRESSION,
+                "zstd",
+                AVRO_COMPRESSION,
+                "zstd",
+                AVRO_COMPRESSION_LEVEL,
+                "9",
+                DELETE_AVRO_COMPRESSION_LEVEL,
+                "9")
+          },
+          {
```

Comment:
Data file properties override the delete file properties when we don't set delete file properties.